src/core/thread/indirect_sender.cpp

```diff
@@ -528,24 +579,37 @@ void IndirectSender::HandleSentFrameToChild(const Mac::TxFrame &aFrame,
             }
         }
 
-        if (message->GetIndirectTxChildMask().Has(childIndex))
+#if OPENTHREAD_FTD
+        if (child != nullptr)
         {
-            message->GetIndirectTxChildMask().Remove(childIndex);
-            mSourceMatchController.DecrementMessageCount(aChild);
+            uint16_t childIndex = Get<ChildTable>().GetChildIndex(*child);
+
+            if (message->GetIndirectTxChildMask().Has(childIndex))
+            {
+                message->GetIndirectTxChildMask().Remove(childIndex);
+                mSourceMatchController.DecrementMessageCount(*child);
+            }
         }
+#endif
 
         Get<MeshForwarder>().RemoveMessageIfNoPendingTx(*message);
     }
 
-    UpdateIndirectMessage(aChild);
+    UpdateIndirectMessage(aNeighbor);
 
 exit:
-    if (mEnabled)
+#if OPENTHREAD_FTD
+    if (mEnabled && (child != nullptr))
     {
         ClearMessagesForRemovedChildren();
     }
+#endif
+
+    return;
```

Comment:
Can remove `return`?

Reply:
I'm afraid its needed for this build:
`OT_CMAKE_NINJA_TARGET="ot-cli-mtd"  ./script/cmake-build simulation -DOT_WAKEUP_END_DEVICE=1`

Otherwise this error is thrown:
`error: label at end of compound statement only available with ‘-std=c++2b’ or ‘-std=gnu++2b’ [-Wc++23-extensions]`

Reply:
Ah! I see. Thanks. Let's keep it.